core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -1604,6 +1604,48 @@ private void checkExponentialCnf(int n) {
         "true");
   }
 
+  /** Unit test for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5780">[CALCITE-5780]
+   * Always-true OR expressions contain reverse order comparison
+   * should be simplified to TRUE</a>. */
+  @Test void testSimplifyOrTermsWithReverseOrderComparison() {
+    final RelDataType intType = typeFactory.createSqlType(SqlTypeName.INTEGER);
+    final RelDataType rowType = typeFactory.builder()
```

Comment:
According to the code change, Maybe `a is not null` should be covered too.

Reply:
@NobiGo IMO, `a is not null` is not releated to the issue we want to solve here, unlike `a is null`, `1 < a or 1 >= a or a is not null` will return UNKNOWN when a's type is nullable and a's value is null, return TRUE when a'type is not nullable.